skill-tree/src/main/java/com/RDS/skilltree/User/UserRole.java

```diff
@@ -1,8 +1,30 @@
 package com.RDS.skilltree.User;
 
-public enum UserRole {
-    USER,
-    MEMBER,
-    SUPERUSER,
-    MAVEN
+import org.springframework.security.core.GrantedAuthority;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public enum UserRole  {
+    USER("user"),
+    MEMBER("member"),
+    SUPERUSER("super_user");
+
+    public final String label;
+
+     UserRole(String label) {
+        this.label = label;
+    }
+    public static UserRole fromString(String text) {
```

Comment:
The fromString method returns null if no matching role is found. While this is a valid approach, consider adding a custom exception or throwing an IllegalArgumentException if a user tries to convert an invalid string to a UserRole. This can improve code clarity and error handling.

Reply:
added the guest role with read-only as we discussed